examples/upsert-resource-at-path/.fn-config/fn-config.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: myService
+  namespace: mySpace
+  annotations:
+    config.kubernetes.io/upsert-path: 'subpkg/service.yaml'
```

Comment:
I don't think this annotation should be named specifically for `upsert`, since if/when we add other functions for modifying resources directly (ex. `patch`) I'd expect them to use the same annotation.

Maybe `target-path`?